docs/overview/architecture.md

```diff
@@ -113,4 +113,28 @@ When the user searches a vector from Vald:
 
 <!-- ### Update -->
 
-<!-- ### Delete -->
+<!-- ### Upsert -->
+
+### Delete
+
+<img src="../../assets/docs/delete_flow.png" />
+
+When the user delete a vector which is indexed in Vald Agent:
+
+1. Vald Ingress receives the delete request from the user. The request includes the vector ID(s) which is specified by the user.
+2. Vald Ingress will forward the request to the Vald Filter Gateway.
+3. Vald Filter Gateway will forward the request to the Vald Meta Gateway. Vald Meta Gateway is used to resolve the internal used UUID of the vector from the Vald Meta.
+4. Vald Meta Gateway will forward the request to the Vald Meta to confirm whether the metadata, which contains the request ID(s), is exist or not.
+5. Vald Meta get the UUID(s) by the request ID(s). Return error if no UUID(s) is found, return with the error.
+6. If Vald Meta Gateway gets the UUID(s), Vald Meta Gateway will forward to the request with the UUID(s) to the Vald LB Gateway through the Vald Backup Gateway.
+7. Vald LB Gateway will broadcast the request with UUID(s) to the Vald Agents. Each Vald Agent will delete the vector data and the medatada if the corresponding UUID(s) is found in the onmemory graph index.
```

Comment:
```suggestion
7. Vald LB Gateway will broadcast the request with UUID(s) to the Vald Agents. Each Vald Agent will delete the vector data and the metadata if the corresponding UUID(s) is found in the on memory graph index.
```

Reply:
@kevindiu  I think `on memory` is not correct grammar 🤔

Reply:
```suggestion
7. Vald LB Gateway will broadcast the request with UUID(s) to the Vald Agents. Each Vald Agent will delete the vector data and the metadata if the corresponding UUID(s) is found in the on-memory graph index.
```
how about it?

Reply:
may be `in memory`?

Reply:
@vankichi `in-memory` should be correct.
https://en.wikipedia.org/wiki/In-memory_database

Reply:
maybe typographical issues 😅 
I'm not sure what is correct...

Reply:
I googled  `on memory graph index`  and seems no one use `on memory`. 
most of the result use `in-memory`

Reply:
```suggestion
7. Vald LB Gateway will broadcast the request with UUID(s) to the Vald Agents. Each Vald Agent will delete the vector data and the metadata if the corresponding UUID(s) is found in the in-memory graph index.
```
Good?

Reply:
I think it is good